projects/auth0-angular/src/lib/auth.client.ts

```diff
@@ -1,18 +1,35 @@
-import { InjectionToken, VERSION } from '@angular/core';
+import { Injectable, VERSION } from '@angular/core';
 import { Auth0Client } from '@auth0/auth0-spa-js';
-import { AuthClientConfig } from './auth.config';
+import { AuthClientConfig, AuthConfig } from './auth.config';
 import useragent from '../useragent';
+import { Observable, race, throwError, timer } from 'rxjs';
+import { map, mergeMap, shareReplay, take } from 'rxjs/operators';
 
-export class Auth0ClientFactory {
-  static createClient(configFactory: AuthClientConfig): Auth0Client {
-    const config = configFactory.get();
+@Injectable()
+export class AuthClient {
+  private instance$ = race(
+    this.config.config$.pipe(
+      map((config) => this.createClient(config)),
+      shareReplay(1)
+    ),
+    timer(30000).pipe(
+      mergeMap(() =>
+        throwError(
+          new Error(
+            'Configuration must be specified either through AuthModule.forRoot or through AuthClientConfig.set'
+          )
+        )
+      )
+    )
+  );
```

Comment:
This serves as a timeout for providing the config to use for creating the client. If this didn't happen within 30seconds of opening the application, it will throw an error.

Typically, the configuration should be loaded way before that. The only reason this would ever be hit is:

- User forgot to configure the SDK
- The configuration is loaded async and is taking more than 30seconds.

Based on the first reason, we would want to throw an error as soon as possible. Keeping the second reason into account, it's important to not throw an error too soon, to allow for slower configuration loading.

We could decide to make this optional, if this ever appears to be a problem. But a user needing a value more than 30 seconds might need to look for a solution for their problem elsewhere.

Reply:
We added `forceInitialization`, which will force the initialization of the SDK. Doing so will result in a 15sec timeout when no config is provided. When the flag is omitted, or set to true, we instantly provide feedback on the missing config.